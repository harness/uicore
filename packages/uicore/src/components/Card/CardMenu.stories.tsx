/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

/* eslint-disable no-alert */
import React from 'react'
import type { Meta, Story } from '@storybook/react'

import { CardBody, Icon, Layout, Text } from '../..'
import { Title, Subtitle, ArgsTable, Stories, PRIMARY_STORY, Primary, Description } from '@storybook/addon-docs/blocks'
import { Card, CardMenuProps } from '../Card/Card'
import { omit } from 'lodash-es'

import * as BP from '@blueprintjs/core'
export default {
  title: 'Components / CardMenu',

  component: CardBody.Menu,
  subcomponents: { Card },
  parameters: {
    layout: 'centered',
    docs: {
      source: {
        type: 'code'
      },

      page: function PageDescription() {
        return (
          <>
            <Title>CardMenu</Title>
            <Description>{`Card Icon component reuses Card Component and lets you to write body for menu [CD Next Gen](https://harness.atlassian.net/wiki/spaces/UI/pages/668271019/Card)
`}</Description>
            <Subtitle>
              <>
                <pre>
                  <code>{`import { CardBody }  from '@harness/uicore'`}</code>
                </pre>
                <pre>
                  <code>{`interface CardProps {
  elevation: number (0|1|2|3|4);
  interactive: boolean (false);
  selected: boolean (false);
  onClick?: (e: React.MouseEvent<HTMLDivElement>) => void;
}

interface CardMenuProps extends HTMLDivProps {
  menuContent: JSX.Element
  title?: string
  colorIdentifier?: Color
}

// Example
// Card Menu
<Card>
    <CardBody.Menu colorIdentifier={'red'} title="Jenkins Artifact" menuContent={<div>Menu</div>}>
        <Text font="small">Referencing the configuration file</Text>
    </CardBody.Menu>
</Card>`}</code>
                </pre>
              </>
            </Subtitle>

            <Primary />
            <ArgsTable story={PRIMARY_STORY} />

            <Stories />
          </>
        )
      }
    }
  },
  decorators: [Story => <Story />]
} as Meta
export const Basic: Story<CardMenuProps> = args => {
  const {
    colorIdentifier = 'red',
    title = 'Jenkins Artifact',
    menuContent = (
      <BP.Menu>
        <BP.Menu.Item icon="edit" text="Edit" />
        <BP.Menu.Item icon="duplicate" text="Clone" />
        <BP.Menu.Item icon="cross" text="Delee" />
      </BP.Menu>
    )
  } = args
  const argsCopy = omit(args, ['menuContent', 'colorIdentifier', 'title'])
  return (
    <>
      <Layout.Horizontal spacing="large">
        <Card onClick={() => alert('Clicked')}>
          <CardBody.Menu colorIdentifier={colorIdentifier} title={title} menuContent={menuContent} {...argsCopy}>
            <div style={{ height: '200px', width: '150px' }}>
              <Layout.Vertical spacing="large">
                <div>
                  <Icon name="service-jenkins" size={30} />
                  <Text style={{ marginTop: '5px' }} font="small">
                    ToDolist.war
                  </Text>
                </div>
                <div>
                  <Text font="small" style={{ marginBottom: '5px', color: 'var(--grey-350)' }}>
                    GETTING THE ARTEFACT
                  </Text>
                  <Text font="small">Referencing the configuration file</Text>
                </div>
                <div>
                  <hr style={{ margin: '0px 0px 10px', border: '1px dashed var(--grey-350)' }} />
                  <Text font="small" style={{ marginBottom: '5px', color: 'var(--grey-350)' }}>
                    ARTEFACT SOURCE
                  </Text>
                  <Text font="small">Fixed</Text>
                </div>
              </Layout.Vertical>
            </div>
          </CardBody.Menu>
        </Card>
      </Layout.Horizontal>
    </>
  )
}
