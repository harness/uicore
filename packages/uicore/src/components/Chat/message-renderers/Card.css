.container {
  width: 100% !important;
  padding: var(--spacing-medium) !important;
  border: 1px solid var(--ai-purple-200);
  border-radius: 6px;
  .header {
    padding-bottom: var(--spacing-medium);
    border-bottom: 1px solid var(--grey-200);
  }
  .body {
    padding-top: var(--spacing-medium);
  }
}
