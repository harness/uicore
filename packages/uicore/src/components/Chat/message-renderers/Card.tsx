import React from 'react'
import { Color, FontVariation, Layout, Text } from '../../../index'
import css from './Card.css'

export interface CardContent {
  title: string
  body: React.ReactNode
}

interface CardProps {
  content: CardContent
}

const Card: React.FC<CardProps> = ({ content }) => {
  return (
    <Layout.Vertical className={css.container}>
      <div className={css.header}>
        <Text color={Color.AI_PURPLE_600} font={{ variation: FontVariation.CARD_TITLE }}>
          {content.title}
        </Text>
      </div>
      <div className={css.body}>{content.body}</div>
    </Layout.Vertical>
  )
}

export default Card
