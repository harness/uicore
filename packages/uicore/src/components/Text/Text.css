/*
 * Copyright (c) Harness Inc.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

.lineclamp {
  display: -webkit-box !important;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: var(--text-line-clamp);
  overflow: hidden;
  word-break: break-all;
}

.withDocsTooltip {
  display: flex;
  align-items: center;
}

.targetClass {
  display: flex !important;
}
.icon {
  vertical-align: middle;
}
