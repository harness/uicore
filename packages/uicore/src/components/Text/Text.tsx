/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { HTMLAttributes, useRef, useState, useCallback, useMemo } from 'react'
import { debounce } from 'lodash-es'
import cx from 'classnames'
import { StyledProps, styledClasses, omitStyledProps } from '@harness/design-system'
import { styledClass } from '@harness/design-system'
import { OptionalTooltip } from '@harness/design-system'
import { Utils } from '../../core/Utils'
import css from './Text.css'
import { Icon, IconName, IconProps } from '@harness/icons'
import { HarnessDocTooltip } from '../../frameworks/Tooltip/Tooltip'

export interface TextProps extends HTMLAttributes<HTMLDivElement>, StyledProps, OptionalTooltip {
  // When lineClamp is specified, show ... (ellipsis) when text is overflown and show the full text
  // in a tooltip on hover. Note that `tooltip` prop takes precedence over this prop.
  // See more: https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-line-clamp
  lineClamp?: number

  /** When true, tooltip will always be shown (even in case lineClamp event is not triggered: no `...` is rendered)  */
  alwaysShowTooltip?: boolean

  /** Left icon */
  icon?: IconName

  /** Optional props for left icon */
  iconProps?: Partial<IconProps>

  /** Right icon */
  rightIcon?: IconName

  /** Optional props for right icon */
  rightIconProps?: Partial<IconProps>

  /** Optional HTML tag. Default is 'p' */
  tag?: string
}

export function Text(props: TextProps): JSX.Element {
  const { icon, iconProps, rightIcon, rightIconProps, alwaysShowTooltip, lineClamp } = props
  const Tag = (props.tag ? props.tag : props.inline ? 'span' : 'p') as React.ElementType
  const [isScrollable, setIsScrollable] = useState(false)
  const ref = useRef<HTMLElement | null>(null)
  const observerRef = useRef<ResizeObserver>()
  const style = { ...props.style }

  if (lineClamp) {
    ;(style as any)['--text-line-clamp'] = lineClamp // eslint-disable-line
  }

  if (icon || rightIcon) {
    ;(style as any)['display'] = props.inline ? 'inline-flex' : 'flex' // eslint-disable-line
    ;(style as any)['alignItems'] = 'center' // eslint-disable-line
  }

  const onResize = useMemo(
    () =>
      debounce(() => {
        if (!ref.current) return

        const element = ref.current
        const heightDifference = (element.scrollHeight || 0) - (element.offsetHeight || 0)
        const widthDifference = (element.scrollWidth || 0) - (element.offsetWidth || 0)

        // Reason to consider difference more than 1 px is
        // If height is 18.2 Px, chrome taking scrollHeight as 18px and firefox take this as 19px
        // where as offsetHeight both take it as 18px
        setIsScrollable(heightDifference > 1 || widthDifference > 1)
      }, 300),
    []
  )

  const refCallback = useCallback(
    (el: HTMLElement | null) => {
      ref.current = el
      observerRef.current?.disconnect()

      if (!ref.current || !lineClamp || !('ResizeObserver' in window)) return
      observerRef.current = new ResizeObserver(onResize)
      observerRef.current.observe(ref.current)
    },
    [lineClamp]
  )

  const tooltip = (() => {
    if (props.tooltip && (alwaysShowTooltip || !lineClamp)) {
      return props.tooltip
    }

    if (isScrollable && lineClamp && lineClamp > 0) {
      return props.tooltip || (props.children as JSX.Element)
    }

    return undefined
  })()

  const wrappedInTooltip = (
    <Utils.WrapOptionalTooltip
      tooltip={tooltip}
      tooltipProps={{
        ...props.tooltipProps,
        targetClassName: cx(css.targetClass, props.tooltipProps?.targetClassName)
      }}>
      <Tag
        {...omitStyledProps(
          Object.assign({}, props, { style }),
          'tooltip',
          'tooltipProps',
          'lineClamp',
          'icon',
          'iconProps',
          'rightIcon',
          'rightIconProps',
          'tag'
        )}
        className={styledClasses(props, styledClass.font, lineClamp && lineClamp >= 1 && css.lineclamp)}
        ref={refCallback}>
        {icon && <Icon className={css.icon} name={icon} size={16} padding={{ right: 'xsmall' }} {...iconProps} />}
        {props.children}
        {rightIcon && (
          <Icon className={css.icon} name={rightIcon} size={16} padding={{ left: 'xsmall' }} {...rightIconProps} />
        )}
      </Tag>
    </Utils.WrapOptionalTooltip>
  )

  return props.tooltipProps?.dataTooltipId ? (
    <div className={css.withDocsTooltip} data-tooltip-id={props.tooltipProps?.dataTooltipId}>
      {wrappedInTooltip}
      <HarnessDocTooltip tooltipId={props.tooltipProps?.dataTooltipId} useStandAlone={true} />
    </div>
  ) : (
    wrappedInTooltip
  )
}

export const SupText: React.FC<TextProps> = props => (
  <sup>
    <Text inline font="xsmall" style={{ padding: '0 2px', marginLeft: 2 }} color="red500" {...props} />
  </sup>
)
