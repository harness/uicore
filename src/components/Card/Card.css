div.card {
  border-radius: 5px;
  display: inline-block;
  position: relative;
  border: 1px solid transparent;

  &:focus {
    box-shadow: 0 0 0 1px rgba(16, 22, 26, 0.1), 0 2px 4px rgba(16, 22, 26, 0.2), 0 8px 24px rgba(16, 22, 26, 0.2);
    outline: none;
  }
}

.interactive {
  &:hover {
    cursor: pointer;
    box-shadow: 0px 0px 2px rgba(40, 41, 61, 0.04), 0px 4px 8px rgba(96, 97, 112, 0.16) !important;
    border: 1px solid var(--blue-400);
  }
}

.cardIcon {
  :global {
    .bp3-icon {
      display: block;
      svg {
        margin: 0 auto var(--spacing-small);
      }
    }
  }
}

.selected {
  border: 2px solid var(--blue-500) !important;
}

.disabled {
  cursor: not-allowed !important;
  background-color: var(--grey-200) !important;
  color: var(--disabled-color) !important;
}

.cardMenu {
  position: relative;
  &.colorIdentifier {
    &:before,
    &:after {
      content: '';
      position: absolute;
      top: -21px;
      left: -21px;
      border-color: transparent;
      border-style: solid;
    }
    &::after {
      border-top-left-radius: 5px;
      border-width: 15px;
      border-left-color: currentColor;
      border-top-color: currentColor;
    }
  }

  .title {
    margin-left: 5px;
    position: relative;
    top: -10px;
    left: 7px;
    width: calc(100% - 20px);
  }

  .dots {
    position: absolute;
    top: -15px;
    right: -19px;

    .menu {
      cursor: pointer;
      outline: none !important;
      color: var(--grey-500);

      &:hover {
        background-color: var(--grey-200) !important;
        color: var(--blue-500);
      }
    }
  }
}

.corner {
  position: absolute;
  top: 0;
  right: 0;
  .badge {
    width: 0;
    position: absolute;
    height: 0;
    border-top: 20px solid var(--blue-500);
    border-left: 20px solid transparent;
    right: 0;
    z-index: 1;
  }
  :global {
    .bp3-icon {
      position: absolute;
      right: 1px;
      top: 1px;
      z-index: 10;
    }
  }
}
