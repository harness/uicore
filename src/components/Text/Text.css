.lineclamp {
  display: -webkit-box !important;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: var(--text-line-clamp);
  overflow: hidden;
  word-break: break-all;
}

.withDocsTooltip {
  display: flex;
  align-items: center;
}

.targetClass {
  display: flex !important;
}
.icon {
  vertical-align: middle;
}
