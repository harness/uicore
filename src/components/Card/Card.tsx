import React, { useCallback } from 'react'
import { Card as BpCard, ICardProps, HTMLDivProps, Position } from '@blueprintjs/core'
import css from './Card.css'
import cx from 'classnames'
import { Icon as UiIcon, IconName, IconProps } from '../../icons/Icon'
import { Color } from '../../core/Color'
import { Popover, PopoverProps } from '../../components/Popover/Popover'
import { Text } from '../../components/Text/Text'
import { Button } from '../../components/Button/Button'

export interface CardProps extends ICardProps {
  selected?: boolean
  disabled?: boolean
  cornerSelected?: boolean
}

const renderCornerTick = () => {
  return (
    <span className={css.corner}>
      <span className={css.badge}></span>
      <UiIcon name="tick" size={9} color={'white'} />
    </span>
  )
}

export const Card: React.FC<CardProps> = props => {
  const {
    selected = false,
    disabled = false,
    className = '',
    style,
    cornerSelected = false,
    tabIndex = 0,
    ...bpProps
  } = props
  // if card is disabled then interactivity should be false
  if (disabled) {
    bpProps.interactive = false
  }
  const tabIndexProps = bpProps.interactive ? { tabIndex } : {}
  const onClick = useCallback(
    event => {
      if (!event.target.closest(`.${css.dots}`)) {
        props.onClick?.(event)
      }
    },
    [props.onClick]
  )

  return (
    <BpCard
      {...bpProps}
      {...tabIndexProps}
      className={cx(
        css.card,
        className,
        { [css.selected]: selected },
        { [css.disabled]: disabled },
        { [css.interactive]: bpProps.interactive }
      )}
      style={style}
      onClick={onClick}>
      {selected && cornerSelected && renderCornerTick()}
      {props.children}
    </BpCard>
  )
}

// Card Body
interface CardIconProps extends HTMLDivProps {
  icon: IconName
  iconSize?: number
  iconProps?: Partial<IconProps>
}

export const Icon: React.FC<CardIconProps> = props => {
  const { icon, iconSize = 16, iconProps = {}, className, style } = props

  return (
    <div className={cx(css.cardIcon, className)} style={style}>
      <UiIcon name={icon} size={iconSize} {...iconProps} />
      {props.children}
    </div>
  )
}

interface CardMenuProps extends HTMLDivProps {
  menuContent: JSX.Element
  title?: string
  colorIdentifier?: Color
  menuPopoverProps?: PopoverProps
}

export const Menu: React.FC<CardMenuProps> = props => {
  const { menuContent, title, colorIdentifier = false, className, style = {}, menuPopoverProps } = props
  // NextJS does not work well with usePortal={true}
  const isNext =
    typeof window !== 'undefined' && typeof window.next !== 'undefined' && typeof window.__NEXT_DATA__ !== 'undefined'

  return (
    <div
      className={cx(css.cardMenu, className, { [css.colorIdentifier]: colorIdentifier })}
      style={{ color: colorIdentifier ? colorIdentifier : '', ...style }}>
      <div className={css.dots}>
        <Popover content={menuContent} position={Position.RIGHT_TOP} usePortal={!isNext} {...menuPopoverProps}>
          <Button className={css.menu} minimal round icon="Options" />
        </Popover>
      </div>
      {title && (
        <Text className={css.title} font="medium">
          {title}
        </Text>
      )}
      {props.children}
    </div>
  )
}

export const CardBody = { Icon, Menu }
