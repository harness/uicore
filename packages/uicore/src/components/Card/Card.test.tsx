/*
 * Copyright 2020 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render, fireEvent, wait } from '@testing-library/react'
import { Card, CardBody } from './Card'
import { Text } from '../../components/Text/Text'

const content = (
  <Text style={{ marginTop: '5px' }} font="medium">
    Jenkin Artifact
  </Text>
)

const iconCardBody = (
  <CardBody.Icon icon="service-kubernetes" iconSize={25}>
    {content}
  </CardBody.Icon>
)

const menuCardBody = (
  <CardBody.Menu colorIdentifier={'red'} title="Jenkins Artifact" menuContent={<div>Menu</div>}>
    {content}
  </CardBody.Menu>
)

describe('Card test', () => {
  test('should render cards with no props', () => {
    const { container } = render(<Card>{content}</Card>)
    expect(container).toMatchSnapshot()
  })
  test('should render cards with props selected and interactive', () => {
    const { container } = render(
      <Card selected={true} interactive={true}>
        {content}
      </Card>
    )
    expect(container).toMatchSnapshot()
  })
  test('should render cards click handler', async () => {
    const onClick = jest.fn()
    const { container } = render(
      <Card selected={true} interactive={true} onClick={onClick}>
        {content}
      </Card>
    )
    fireEvent.click(container.children[0])
    await wait()
    expect(onClick).toHaveBeenCalled()
  })
})

describe('Card icon test', () => {
  test('should render icon card with no props', () => {
    const { container } = render(<Card>{iconCardBody}</Card>)
    expect(container).toMatchSnapshot()
  })
  test('should render cards with props selected and interactive', () => {
    const { container } = render(
      <Card selected={true} interactive={true}>
        {iconCardBody}
      </Card>
    )
    expect(container).toMatchSnapshot()
  })
  test('should render cards click handler', async () => {
    const onClick = jest.fn()
    const { container } = render(
      <Card selected={true} interactive={true} onClick={onClick}>
        {iconCardBody}
      </Card>
    )
    fireEvent.click(container.children[0])
    await wait()
    expect(onClick).toHaveBeenCalled()
  })
})

describe('Card menu test', () => {
  test('should render icon card with no props', () => {
    const { container } = render(<Card>{menuCardBody}</Card>)
    expect(container).toMatchSnapshot()
  })
  test('should render cards with props selected and interactive', () => {
    const { container } = render(
      <Card selected={true} interactive={true}>
        {menuCardBody}
      </Card>
    )
    expect(container).toMatchSnapshot()
  })
  test('should render cards click handler', async () => {
    const onClick = jest.fn()
    const { container } = render(
      <Card selected={true} interactive={true} onClick={onClick}>
        {menuCardBody}
      </Card>
    )
    fireEvent.click(container.children[0])
    await wait()
    expect(onClick).toHaveBeenCalled()
  })
})
