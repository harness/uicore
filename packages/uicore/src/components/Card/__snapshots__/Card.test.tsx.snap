// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card icon test should render cards with props selected and interactive 1`] = `
<div>
  <div
    class="bp3-card bp3-interactive bp3-elevation-0 card selected interactive"
    tabindex="0"
  >
    <div
      class="cardIcon"
    >
      <span
        class="bp3-icon StyledProps--main"
        data-icon="service-kubernetes"
      >
        <svg
          height="25"
          viewBox="0 0 14 14"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 .4c-.1 0-.2 0-.3.1L2 2.7c-.2.1-.4.3-.5.6l-1.2 5c-.1.2 0 .5.1.7l.1.1 3.2 4c.2.2.4.3.7.3h5.2c.3 0 .5-.1.7-.3l3.2-4c.2-.2.2-.5.2-.7l-1.2-5c-.1-.3-.2-.5-.5-.6L7.3.6c0-.1-.2-.2-.3-.2z"
            fill="#326ce5"
          />
          <path
            d="M7 2.1c-.2 0-.3.1-.3.3v.9c0 .1-.1.1-.1.1v.1c-.2 0-.3 0-.5.1-.7.2-1.4.5-1.8 1l-.1-.1H4c-.1-.1-.2-.2-.4-.3L3.4 4h-.1c-.1 0-.1-.1-.2-.1s-.2 0-.2.1c-.1.1-.1.3.1.4h.1c.1.1.2.1.2.1.2.1.3.2.4.3 0 0 .1.1.1.2l.1.1c-.5.7-.7 1.6-.6 2.5h-.1l-.1.1c-.1 0-.3.1-.5.1h-.4c-.2 0-.3.2-.2.3 0 .1.2.2.4.2h.1c.1 0 .2-.1.3-.1.2-.1.3-.1.5-.1.1 0 .1 0 .2.1h.1c.3.9.9 1.6 1.6 2l-.1.1v.2c-.1.1-.1.3-.3.5-.1.1-.1.1-.1.2v.1c-.1.2 0 .3.1.4s.3 0 .4-.2v-.1c0-.1.1-.2.1-.3.1-.2.1-.4.2-.5 0 0 .1 0 .1-.1l.1-.1c.6.2 1.4.3 2.1.2.2 0 .3-.1.5-.1l.1.1s.1 0 .1.1c.1.1.1.3.2.5 0 .1 0 .2.1.3v.1c.1.2.2.2.4.2.1-.1.2-.2.1-.4v-.1c0-.1-.1-.1-.1-.2-.1-.2-.2-.3-.2-.4V10.4c.8-.5 1.3-1.2 1.6-2h.1s.1-.1.2-.1.3.1.5.1c.1 0 .2.1.3.1h.1c.2 0 .3-.1.4-.2 0-.1-.1-.3-.2-.3h-.4c-.2 0-.4 0-.5-.1-.1 0-.1-.1-.1-.1h-.1c.1-.4 0-.9-.1-1.3s-.3-.9-.5-1.2l.1-.1c0-.1 0-.1.1-.2s.2-.2.4-.3c.1 0 .2-.1.2-.1h.1c.1-.1.2-.3.1-.4s-.3-.1-.4 0h-.1l-.2.2c-.1.1-.3.3-.4.3H10l-.1.1c-.6-.6-1.4-1-2.3-1.1v-.1l-.1-.1V3v-.4c-.2-.3-.3-.5-.5-.5zm-.3 2.2l-.1 1.5c0 .1-.1.2-.2.2h-.1l-1.2-.9c.4-.4.8-.6 1.4-.8zm.7 0c.6.1 1.2.4 1.7.8l-1.2.8c-.1.1-.3.1-.3 0l-.1-.1zM4.6 5.7l1.1 1c.1.1.1.2 0 .3l-.1.1-1.4.4c-.1-.7 0-1.3.4-1.8zm4.9 0c.2.3.3.6.4.9s.1.6.1.9l-1.4-.4c-.1 0-.2-.2-.2-.3s0-.1.1-.1zm-2.7 1h.4l.3.3-.1.4-.4.2-.4-.2-.1-.4zm1.4 1.2c.1 0 .1 0 0 0l1.5.2c-.2.6-.6 1.1-1.2 1.5l-.6-1.4c.1-.1.1-.2.3-.3zm-2.4 0c.1 0 .2.1.2.2v.2l-.6 1.4c-.5-.3-.9-.8-1.2-1.5zm1.2.6l.2.1.7 1.3c-.1 0-.2.1-.3.1-.5.1-1.1.1-1.6-.1l.7-1.3c.2 0 .2-.1.3-.1z"
            fill="#fff"
            stroke="#fff"
            stroke-width="0.25"
          />
        </svg>
      </span>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-medium"
        style="margin-top: 5px;"
      >
        Jenkin Artifact
      </p>
    </div>
  </div>
</div>
`;

exports[`Card icon test should render icon card with no props 1`] = `
<div>
  <div
    class="bp3-card bp3-elevation-0 card"
  >
    <div
      class="cardIcon"
    >
      <span
        class="bp3-icon StyledProps--main"
        data-icon="service-kubernetes"
      >
        <svg
          height="25"
          viewBox="0 0 14 14"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 .4c-.1 0-.2 0-.3.1L2 2.7c-.2.1-.4.3-.5.6l-1.2 5c-.1.2 0 .5.1.7l.1.1 3.2 4c.2.2.4.3.7.3h5.2c.3 0 .5-.1.7-.3l3.2-4c.2-.2.2-.5.2-.7l-1.2-5c-.1-.3-.2-.5-.5-.6L7.3.6c0-.1-.2-.2-.3-.2z"
            fill="#326ce5"
          />
          <path
            d="M7 2.1c-.2 0-.3.1-.3.3v.9c0 .1-.1.1-.1.1v.1c-.2 0-.3 0-.5.1-.7.2-1.4.5-1.8 1l-.1-.1H4c-.1-.1-.2-.2-.4-.3L3.4 4h-.1c-.1 0-.1-.1-.2-.1s-.2 0-.2.1c-.1.1-.1.3.1.4h.1c.1.1.2.1.2.1.2.1.3.2.4.3 0 0 .1.1.1.2l.1.1c-.5.7-.7 1.6-.6 2.5h-.1l-.1.1c-.1 0-.3.1-.5.1h-.4c-.2 0-.3.2-.2.3 0 .1.2.2.4.2h.1c.1 0 .2-.1.3-.1.2-.1.3-.1.5-.1.1 0 .1 0 .2.1h.1c.3.9.9 1.6 1.6 2l-.1.1v.2c-.1.1-.1.3-.3.5-.1.1-.1.1-.1.2v.1c-.1.2 0 .3.1.4s.3 0 .4-.2v-.1c0-.1.1-.2.1-.3.1-.2.1-.4.2-.5 0 0 .1 0 .1-.1l.1-.1c.6.2 1.4.3 2.1.2.2 0 .3-.1.5-.1l.1.1s.1 0 .1.1c.1.1.1.3.2.5 0 .1 0 .2.1.3v.1c.1.2.2.2.4.2.1-.1.2-.2.1-.4v-.1c0-.1-.1-.1-.1-.2-.1-.2-.2-.3-.2-.4V10.4c.8-.5 1.3-1.2 1.6-2h.1s.1-.1.2-.1.3.1.5.1c.1 0 .2.1.3.1h.1c.2 0 .3-.1.4-.2 0-.1-.1-.3-.2-.3h-.4c-.2 0-.4 0-.5-.1-.1 0-.1-.1-.1-.1h-.1c.1-.4 0-.9-.1-1.3s-.3-.9-.5-1.2l.1-.1c0-.1 0-.1.1-.2s.2-.2.4-.3c.1 0 .2-.1.2-.1h.1c.1-.1.2-.3.1-.4s-.3-.1-.4 0h-.1l-.2.2c-.1.1-.3.3-.4.3H10l-.1.1c-.6-.6-1.4-1-2.3-1.1v-.1l-.1-.1V3v-.4c-.2-.3-.3-.5-.5-.5zm-.3 2.2l-.1 1.5c0 .1-.1.2-.2.2h-.1l-1.2-.9c.4-.4.8-.6 1.4-.8zm.7 0c.6.1 1.2.4 1.7.8l-1.2.8c-.1.1-.3.1-.3 0l-.1-.1zM4.6 5.7l1.1 1c.1.1.1.2 0 .3l-.1.1-1.4.4c-.1-.7 0-1.3.4-1.8zm4.9 0c.2.3.3.6.4.9s.1.6.1.9l-1.4-.4c-.1 0-.2-.2-.2-.3s0-.1.1-.1zm-2.7 1h.4l.3.3-.1.4-.4.2-.4-.2-.1-.4zm1.4 1.2c.1 0 .1 0 0 0l1.5.2c-.2.6-.6 1.1-1.2 1.5l-.6-1.4c.1-.1.1-.2.3-.3zm-2.4 0c.1 0 .2.1.2.2v.2l-.6 1.4c-.5-.3-.9-.8-1.2-1.5zm1.2.6l.2.1.7 1.3c-.1 0-.2.1-.3.1-.5.1-1.1.1-1.6-.1l.7-1.3c.2 0 .2-.1.3-.1z"
            fill="#fff"
            stroke="#fff"
            stroke-width="0.25"
          />
        </svg>
      </span>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-medium"
        style="margin-top: 5px;"
      >
        Jenkin Artifact
      </p>
    </div>
  </div>
</div>
`;

exports[`Card menu test should render cards with props selected and interactive 1`] = `
<div>
  <div
    class="bp3-card bp3-interactive bp3-elevation-0 card selected interactive"
    tabindex="0"
  >
    <div
      class="cardMenu colorIdentifier"
      style="color: red;"
    >
      <div
        class="dots"
      >
        <span
          class="bp3-popover-wrapper"
        >
          <span
            class="bp3-popover-target"
          >
            <button
              class="bp3-button bp3-minimal button StyledProps--font StyledProps--main menu with-current-color round iconOnly without-shadow withLeftIcon"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-more StyledProps--main"
                icon="more"
              >
                <svg
                  data-icon="more"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    more
                  </desc>
                  <path
                    d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </span>
        </span>
      </div>
      <p
        class="StyledProps--font StyledProps--main title StyledProps--font-medium"
      >
        Jenkins Artifact
      </p>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-medium"
        style="margin-top: 5px;"
      >
        Jenkin Artifact
      </p>
    </div>
  </div>
</div>
`;

exports[`Card menu test should render icon card with no props 1`] = `
<div>
  <div
    class="bp3-card bp3-elevation-0 card"
  >
    <div
      class="cardMenu colorIdentifier"
      style="color: red;"
    >
      <div
        class="dots"
      >
        <span
          class="bp3-popover-wrapper"
        >
          <span
            class="bp3-popover-target"
          >
            <button
              class="bp3-button bp3-minimal button StyledProps--font StyledProps--main menu with-current-color round iconOnly without-shadow withLeftIcon"
              type="button"
            >
              <span
                class="bp3-icon bp3-icon-more StyledProps--main"
                icon="more"
              >
                <svg
                  data-icon="more"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <desc>
                    more
                  </desc>
                  <path
                    d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
                    fill-rule="evenodd"
                  />
                </svg>
              </span>
            </button>
          </span>
        </span>
      </div>
      <p
        class="StyledProps--font StyledProps--main title StyledProps--font-medium"
      >
        Jenkins Artifact
      </p>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-medium"
        style="margin-top: 5px;"
      >
        Jenkin Artifact
      </p>
    </div>
  </div>
</div>
`;

exports[`Card test should render cards with no props 1`] = `
<div>
  <div
    class="bp3-card bp3-elevation-0 card"
  >
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-medium"
      style="margin-top: 5px;"
    >
      Jenkin Artifact
    </p>
  </div>
</div>
`;

exports[`Card test should render cards with props selected and interactive 1`] = `
<div>
  <div
    class="bp3-card bp3-interactive bp3-elevation-0 card selected interactive"
    tabindex="0"
  >
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-medium"
      style="margin-top: 5px;"
    >
      Jenkin Artifact
    </p>
  </div>
</div>
`;
